internal/service/lightsail/database.go

```diff
@@ -0,0 +1,462 @@
+package lightsail
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func ResourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: ResourceDatabaseCreate,
+		Read:   ResourceDatabaseRead,
+		Update: ResourceDatabaseUpdate,
+		Delete: ResourceDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: ResourceDatabaseImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
Is this meant to meet the `RelationalDatabaseName` or the `MasterDatabaseName` parameter of the input struct? This is not clear to me. Please clarify?

You could mimic how the RDS resource is implemented, but I feel that it adds unnecessary complexity with conditional evaluation and [ConflictsWith clauses](https://github.com/hashicorp/terraform-provider-aws/blob/a7f0bd58b9752f8dd7d738239515329c5042c28e/internal/service/rds/instance.go#L301). See:

```
dbName := d.Get("db_name").(string)
if dbName == "" {
	dbName = d.Get("name").(string)
}

```

Reference:
https://github.com/hashicorp/terraform-provider-aws/blob/a7f0bd58b9752f8dd7d738239515329c5042c28e/internal/service/rds/instance.go#L714

Reply:
I change the `name` field to `relational_database_name` to be more explicit about what the field was for.